src/figuresClasses.ts

```diff
@@ -1,11 +1,92 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public firstSide: number,
+    public secondSide: number,
+    public thirdSide: number,
+  ) {
+    const biggestSide: number = Math.max(firstSide, secondSide, thirdSide);
+    const otherSidesSum: number = [firstSide, secondSide, thirdSide]
+      .filter((side) => side !== biggestSide)
+      .reduce((acc, side) => acc + side);
+
+    if (firstSide <= 0 || secondSide <= 0 || thirdSide <= 0) {
```

Comment:
[CODE STYLE] - The error message thrown here should be more descriptive to clearly indicate the nature of the problem. Consider providing a message that explains why the sides must be positive, such as 'Triangle sides must be positive numbers'.